src/components/generic/OptionsList/OptionsList.jsx

```diff
@@ -27,16 +28,34 @@ export default function OptionsList({
   withKey,
 }) {
   const [cssClasses, sc] = webexComponentClasses('options-list', className);
+  const optionsListRef = useRef();
+  const elementPosition = useElementPosition(optionsListRef);
+  const [maxHeight, setMaxHeight] = useState(0);
+
+  const handleSetSize = () => {
+    if (elementPosition) {
+      setMaxHeight(window.innerHeight - elementPosition.top - window.scrollY - 50);
+    }
+  };
 
   const onKeyDown = (event) => {
     if (event.key === 'Tab') {
       onBlur();
     }
   };
 
+  useEffect(() => {
+    const cleanup = () => window.removeEventListener('resize', handleSetSize);
+
+    handleSetSize();
+    window.addEventListener('resize', handleSetSize);
+
+    return cleanup;
+  });
```

Comment:
if no deps, this effect will execute after each render.

The registering of the resize listener should be done only once, and I think it would be better to let useElementPosition handle it.